internal/ingress/controller/template/template.go

```diff
@@ -1013,13 +1014,29 @@ func buildOpentracing(c interface{}, s interface{}) string {
 	}
 
 	buf := bytes.NewBufferString("")
+	var host string = "localhost"
 
 	if cfg.DatadogCollectorHost != "" {
 		buf.WriteString("opentracing_load_tracer /usr/local/lib64/libdd_opentracing.so /etc/nginx/opentracing.json;")
+		host = cfg.DatadogCollectorHost
 	} else if cfg.ZipkinCollectorHost != "" {
 		buf.WriteString("opentracing_load_tracer /usr/local/lib/libzipkin_opentracing_plugin.so /etc/nginx/opentracing.json;")
+		host = cfg.ZipkinCollectorHost
 	} else if cfg.JaegerCollectorHost != "" || cfg.JaegerEndpoint != "" {
 		buf.WriteString("opentracing_load_tracer /usr/local/lib/libjaegertracing_plugin.so /etc/nginx/opentracing.json;")
+		host = cfg.JaegerCollectorHost
```

Comment:
don't we care about resolution of `JaegerEndpoint`?

Reply:
oops, bad auto-resolve after rebase. This PR isn't ready since the latest changes to master, I'll do another once-over